src/nodes.coffee

```diff
@@ -961,16 +961,25 @@ exports.StringLiteral = class StringLiteral extends Literal
       @double
     }
 
+    @unquotedValueForTemplateLiteral = makeDelimitedLiteral val, {
```

Comment:
Again, we get into the territory of actual-string-literal vs text-chunk-of-interpolated-string

I'm not sure that's fully untangled yet but I think this is an improvement - rather than always quoting the value here in `StringLiteral::constructor()` and then undoing that quoting in a couple places (in a mutating way), it creates the different unquoted variants it may need here and then uses those later as needed. I got rid of `StringLiteral::unquote()` entirely